aten/src/ATen/native/TensorShape.cpp

```diff
@@ -545,6 +545,21 @@ std::vector<Tensor> tensor_split(const Tensor& self, IntArrayRef indices, int64_
   return splits;
 }
 
+std::vector<Tensor> tensor_split(const Tensor& self, const Tensor& tensor_indices_or_sections, int64_t dim) {
```

Comment:
please check that `indices_or_sections` is on cpu, it won't work otherwise (first branch will cause host-device sync, second branch will segfault when you are creating std::vector from tensor data)

Reply:
Hi @ngimel , how would the code look like to check `indices_or_sections` is on cpu? Is this a TORCH_CHECK?
Once `indices_or_sections` is confirmed to be on cpu does the body of the code change?

I think we want to support this for gpu as well, how would the code change to support this?
cc. @xw285cornell

Reply:
Yes, it should be a TORCH_CHECK, something like
```
TORCH_CHECK(tensor_indices_or_sections.device()==kCPU, "expected tensor_indices_or_sections to be on cpu")
```
After the check the body of the code does not need to change. Allowing `indices_or_sections` to be on gpu would require you to copy it to cpu here, which is a cross-device synchronization. Pytorch tries not to do implicit cross-device synchronization, if you need it, you can do it in the client code by calling `indices.cpu()`